velox/functions/lib/Re2Functions.cpp

```diff
@@ -561,7 +615,7 @@ class LikeGeneric final : public VectorFunction {
       EvalCtx& context,
       VectorPtr& result) const final {
     VectorPtr localResult;
-
+    bool isAscii = isAsciiArg(rows, args[0]);
```

Comment:
This is needed only for xxxN patterns. Perhaps, refactor to avoid computing this for all cases.

Reply:
@mbasmanova LikeGeneric process input row by row, which means every input might have different pattern, which makes us hard to refactor to only calculate ASCIIness when the pattern is kxxxN?

Reply:
Got it. Thank you for clarifying.